okhttp-client-handler/src/main/scala/com/softwaremill/sttp/okhttp/OkHttpClientHandler.scala

```diff
@@ -0,0 +1,132 @@
+package com.softwaremill.sttp.okhttp
+
+import java.io.IOException
+import java.nio.charset.Charset
+
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.model._
+import okhttp3.internal.http.HttpMethod
+import okhttp3.{
+  Call,
+  Callback,
+  MediaType,
+  OkHttpClient,
+  Request => OkHttpRequest,
+  RequestBody => OkHttpRequestBody,
+  Response => OkHttpResponse
+}
+import okio.{BufferedSink, Okio}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Future, Promise}
+import scala.language.higherKinds
+
+abstract class OkHttpClientHandler[R[_], S](client: OkHttpClient)
+    extends SttpHandler[R, S] {
+  private[okhttp] def convertRequest[T](request: Request[T, S]): OkHttpRequest = {
+    val builder = new OkHttpRequest.Builder()
+      .url(request.uri.toString)
+
+    val body = setBody(request.body)
+    builder.method(request.method.m, body.getOrElse {
+      if (HttpMethod.requiresRequestBody(request.method.m))
+        OkHttpRequestBody.create(null, "")
+      else null
+    })
+
+    //OkHttp support automatic gzip compression
+    request.headers.filter(_._1 != "Accept-Encoding").foreach {
```

Comment:
In what sense the gzip support is automatic? Does it add the `Accept-Encoding: gzip, deflate` header by default? Is it an error to specify the header yourself?

As for the actual filtering, I think it would be better to do `_._1.eqalsIgnoreCase(AcceptEncodingHeader)` as (1) you never know what case the user is going to specify the header in and (2) we can utilize a constant